### ISSUE TITLE
NPE generating latticegraph

### ISSUE THREAD

**Issue by sbliven** [eppic-team/eppic#174]
This gives an error:
http://www.eppic-web.org:8091/ewui/ewui/latticeGraph?id=5t89&interfaces=3,1,5-8&size=500

Happily, we give a reasonable error message to the user! Less happily, we log the null pointer:

```
2017-03-15 17:28:14 [pool-2-thread-2] ERROR ch.systemsx.sybit.crkwebui.server.commons.util.io.FileCache - Error writing /data/webapps/files_2017_01/data/divided/t8/5t89/5t89.latticeGraph.1,3,5-8.json
java.util.concurrent.ExecutionException: java.lang.NullPointerException
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_121]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_121]
        at ch.systemsx.sybit.crkwebui.server.commons.util.io.FileCache$CacheFile.lambda$new$1(FileCache.java:86) ~[classes/:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
        at org.biojava.nbio.structure.contact.StructureInterface.getContactOverlapScore(StructureInterface.java:597) ~[biojava-structure-4.2.6.jar:4.2.6]
        at org.biojava.nbio.structure.contact.StructureInterface.isIsologous(StructureInterface.java:659) ~[biojava-structure-4.2.6.jar:4.2.6]
        at eppic.assembly.InterfaceEdge.setInterface(InterfaceEdge.java:84) ~[eppic-cli-3.0-RC2-SNAPSHOT.jar:?]
        at eppic.assembly.LatticeGraph.initLatticeGraphTopologically(LatticeGraph.java:337) ~[eppic-cli-3.0-RC2-SNAPSHOT.jar:?]
        at eppic.assembly.LatticeGraph.<init>(LatticeGraph.java:116) ~[eppic-cli-3.0-RC2-SNAPSHOT.jar:?]
        at eppic.assembly.LatticeGraph3D.<init>(LatticeGraph3D.java:86) ~[eppic-cli-3.0-RC2-SNAPSHOT.jar:?]
        at ch.systemsx.sybit.crkwebui.server.jmol.generators.LatticeGraphPageGenerator.lambda$generateJSONPage$0(LatticeGraphPageGenerator.java:135) ~[classes/:?]
        ... 4 more
2017-03-15 17:28:14 [qtp410424423-30] ERROR ch.systemsx.sybit.crkwebui.server.jmol.servlets.LatticeGraphServlet - Error during preparation of Lattice Graph page.
java.util.concurrent.ExecutionException: java.lang.NullPointerException
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_121]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_121]
        at ch.systemsx.sybit.crkwebui.server.commons.util.io.FileCache$CacheFile.get(FileCache.java:111) ~[classes/:?]
        at ch.systemsx.sybit.crkwebui.server.commons.util.io.FileCache.getString(FileCache.java:299) ~[classes/:?]
        at ch.systemsx.sybit.crkwebui.server.jmol.generators.LatticeGraphPageGenerator.generateJSONPage(LatticeGraphPageGenerator.java:148) ~[classes/:?]
        at ch.systemsx.sybit.crkwebui.server.jmol.servlets.LatticeGraphServlet.doGet(LatticeGraphServlet.java:156) [classes/:?]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:687) [servlet-api-3.1.jar:3.1.0]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [servlet-api-3.1.jar:3.1.0]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812) [jetty-servlet-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669) [jetty-servlet-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.servlets.CrossOriginFilter.handle(CrossOriginFilter.java:256) [jetty-servlets-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlets.CrossOriginFilter.doFilter(CrossOriginFilter.java:219) [jetty-servlets-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.21.v20170120.jar:9.2.21.v20170120]
        at ch.systemsx.sybit.crkwebui.server.crawler.servlets.CrawlerFilter.doFilter(CrawlerFilter.java:58) [classes/:?]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.servlets.UserAgentFilter.doFilter(UserAgentFilter.java:83) [jetty-servlets-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlets.GzipFilter.doFilter(GzipFilter.java:364) [jetty-servlets-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585) [jetty-servlet-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577) [jetty-security-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515) [jetty-servlet-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:215) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:110) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.Server.handle(Server.java:499) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:258) [jetty-server-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544) [jetty-io-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.21.v20170120.jar:9.2.21.v20170120]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.21.v20170120.jar:9.2.21.v20170120]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
        at org.biojava.nbio.structure.contact.StructureInterface.getContactOverlapScore(StructureInterface.java:597) ~[biojava-structure-4.2.6.jar:4.2.6]
        at org.biojava.nbio.structure.contact.StructureInterface.isIsologous(StructureInterface.java:659) ~[biojava-structure-4.2.6.jar:4.2.6]
        at eppic.assembly.InterfaceEdge.setInterface(InterfaceEdge.java:84) ~[eppic-cli-3.0-RC2-SNAPSHOT.jar:?]
        at eppic.assembly.LatticeGraph.initLatticeGraphTopologically(LatticeGraph.java:337) ~[eppic-cli-3.0-RC2-SNAPSHOT.jar:?]
        at eppic.assembly.LatticeGraph.<init>(LatticeGraph.java:116) ~[eppic-cli-3.0-RC2-SNAPSHOT.jar:?]
        at eppic.assembly.LatticeGraph3D.<init>(LatticeGraph3D.java:86) ~[eppic-cli-3.0-RC2-SNAPSHOT.jar:?]
        at ch.systemsx.sybit.crkwebui.server.jmol.generators.LatticeGraphPageGenerator.lambda$generateJSONPage$0(LatticeGraphPageGenerator.java:135) ~[classes/:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_121]
        ... 1 more
```

**Comment by sbliven** [comment]
The assembly diagrams generate, so this isn't a simple permissions problem.

Also effects production

**Comment by lafita** [comment]
Any idea how to solve or debug that?

**Comment by josemduarte** [comment]
The problem here is that for some reason the CLI didn't generate the json data for this entry. We should look into why not. The thing is that the latticeGraph generation on-the-fly got broken at some point (I broke it while trying to fix something else), that's the bug you are seeing in the stack trace.

**Comment by josemduarte** [comment]
See #159 for the underlying issue that causes the NPE

**Comment by josemduarte** [comment]
Now fixed with the fix to #159 . First time it runs it is quite slow, after that it's cached.
